packages/davinci-client/src/lib/client.store.ts

```diff
@@ -166,12 +186,34 @@ export async function davinci<ActionType extends ActionTypes = ActionTypes>({
      * @method: resume - Resume a social login flow when returned to application
      * @returns unknown
      */
-    resume: async ({ continueToken }: { continueToken: string }) => {
-      await store.dispatch(davinciApi.endpoints.resume.initiate({ continueToken }));
-
-      const node = nodeSlice.selectSlice(store.getState());
-
-      return node;
+    resume: async ({
+      continueToken,
+    }: {
+      continueToken: string;
+    }): Promise<InternalErrorResponse | NodeStates> => {
+      try {
+        const serverInfo = await serverStorage.get();
+        if (serverInfo && '_links' in serverInfo) {
```

Comment:
I get that you're narrowing here, but it feels a bit overkill for me since this very same check exists within the API module. Can we just do a cast here, and just check that truthiness of the value? We can then rely on the more strict check within the API module for the more detailed property needs.